### ISSUE TITLE
Feature request: implicitly exclude Kotlin `@Metadata` annotations from being copied

### ISSUE THREAD

**Issue by ZacSweers** [google/auto#1087]
Context: kotlinc generates classes annotated with `@kotlin.Metadata` annotations on them. These annotations contain binary data about the kotlin language features, which kotlin then uses to understand those features in compiled binaries (jars, class files, etc).

These `@Metadata` annotations are not intended to be present on classes other than the one generated by kotlinc, let alone on Java classes. Doing so can confuse or break kotlin tooling and reflection that inspects these annotations. 

AutoValue will automatically copy these annotations though unless opted out via `@CopyAnnotations(excludes = [Metadata::class])`. I'd like to feature request _always_ excluding copying these, as it is always an error to copy them. Happy to send a PR if you're open to it!

**Comment by eamonnmcmanus** [comment]
I'm not averse to doing this, but I'm curious as to why you would be writing AutoValue classes in Kotlin. Aren't Kotlin data classes a better fit?

**Comment by ZacSweers** [comment]
Generally yes but we have some tools built as autovalue extensions that don't have readily-available kotlin solutions yet.

**Comment by ZacSweers** [comment]
I'm attempting a PR for this but the project seems to be essentially unusable in IntelliJ. Is there a recommended local development flow?

![image](https://user-images.githubusercontent.com/1361086/118150306-685e7880-b3e0-11eb-9daa-ade43f14ae52.png)

**Comment by eamonnmcmanus** [comment]
I tried launching IntelliJ on a freshly-created clone of this repo on my Mac with IntelliJ CE 2020.2 (with some Google plugins) and I did see some messages like the one you show here. But I disabled a plugin (BSFConsole) and restarted, and then the messages went away. I was able to run the tests from within the IDE. So possibly your problem is coming from a plugin?
![intellij-auto-value](https://user-images.githubusercontent.com/5246810/118157999-16155b80-b3d0-11eb-8126-bfb169b5e53f.png)
